jbpm-case-mgmt/jbpm-case-mgmt-cmmn/src/main/java/org/jbpm/casemgmt/cmmn/xml/SentryHandler.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright 2018 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.jbpm.casemgmt.cmmn.xml;
+
+import java.util.HashSet;
+import java.util.Map;
+
+import org.drools.core.xml.BaseAbstractHandler;
+import org.drools.core.xml.ExtensibleXmlParser;
+import org.drools.core.xml.Handler;
+import org.jbpm.casemgmt.cmmn.core.PlanItem;
+import org.jbpm.casemgmt.cmmn.core.Sentry;
+import org.jbpm.compiler.xml.ProcessBuildData;
+import org.jbpm.ruleflow.core.RuleFlowProcess;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.w3c.dom.Element;
+import org.xml.sax.Attributes;
+import org.xml.sax.SAXException;
+
+public class SentryHandler extends BaseAbstractHandler implements Handler {
+
+    private static final Logger logger = LoggerFactory.getLogger(SentryHandler.class);
+
+    public SentryHandler() {
+        if ((this.validParents == null) && (this.validPeers == null)) {
+            this.validParents = new HashSet<>();
+            this.validParents.add(RuleFlowProcess.class);
+
+            this.validPeers = new HashSet<>();
+            this.validPeers.add(null);
+            this.validPeers.add(PlanItem.class);
+            this.validPeers.add(Sentry.class);
+
+            this.allowNesting = false;
+        }
```

Comment:
I think this comment has disappeared somehow and I haven't found a reply to it.

Is the ```if ((this.validParents == null) && (this.validPeers == null)) {``` relevant in these handlers? Because by looking at the code I cannot think of case where they are not null. BaseAbstractHandler does not initialize them nor there are any setters to set them. So I believe if we initialized them without checking, we might make it cleaner and quicker as well :) The IF is also omitted in AbstractCaseNodeHandler...

Reply:
in general I simply followed the same as it was done for BPMN2 parsers so wanted to stay as close to it as possible ...

Reply:
OK. So should I understand it that don't touch it if it works? :)

Reply:
yeah, that kind of thing :)